addons/hr_holidays/models/hr_employee.py

```diff
@@ -217,6 +217,18 @@ def write(self, values):
         # remove users from the Responsible group if they are no longer leave managers
         old_managers.sudo()._clean_leave_responsible_users()
 
+        # Change the resource calendar of the employee's leaves in the future
+        if 'resource_calendar_id' in values:
+            try:
+                self.env['hr.leave'].search([
+                    ('employee_id', 'in', self.ids),
+                    ('resource_calendar_id', '!=', int(values['resource_calendar_id'])),
+                    ('date_from', '>', fields.Datetime.now())]).write({'resource_calendar_id': values['resource_calendar_id']})
+            except ValidationError:
+                raise ValidationError(_("Changing this working schedule results in the affected employee(s) not having enough "
+                                        "leaves allocated to accomodate for their leaves already taken in the future. Please "
+                                        "review this employee's leaves and adjust their allocation accordingly."))
```

Comment:
This code is to accommodate the use case of not using contract. But it's true that it will be triggered twice when it comes from contract, once in contract and once here.

Reply:
Fixed